sdk/core/Azure.Core/tests/TransportFunctionalTests.cs

```diff
@@ -917,6 +920,56 @@ public async Task StreamReadingExceptionsAreIOExceptions()
             }
         }
 
+        [Test]
+        public async Task ServerCertificateCustomValidationCallbackIsHonored([Values(true, false)] bool setCertCallback,[Values(true, false)]  bool isValidCert)
+        {
+            using (TestServer testServer = new TestServer(
+                async context =>
+                {
+                    byte[] buffer = Encoding.UTF8.GetBytes("Hello");
+                    await context.Response.Body.WriteAsync(buffer, 0, buffer.Length);
+                }, true))
+            {
+                bool certValidationCalled = false;
+                var options = new HttpPipelineTransportOptions();
+
+                if (setCertCallback)
+                {
+                    options.ServerCertificateCustomValidationCallback = certificate2 =>
+                    {
+                        certValidationCalled = true;
+                        Assert.NotNull(certificate2);
+                        return isValidCert;
```

Comment:
Should we use isValidCert in one of the asserts?

Reply:
thanks, I missed this after refactoring a few times.